TradingBot/Services/Strategy.cs

```diff
@@ -1,33 +1,188 @@
         {
-            TradingAction.MarketBuy(new TradingSymbol("TSLA"), 0.5m, _clock.UtcNow)
-        };
+            var symbol = prediction.Key;
+            var closingPrices = new List<decimal>();
+            var currentPrice = await GetCurrentPrice(symbol, token);
+            if (currentPrice != null)
+                closingPrices.Add(currentPrice.Close);
+
+            closingPrices.AddRange(prediction.Value.Prices.Select(dailyPrice => dailyPrice.ClosingPrice).ToList());    
+
+            if (IsPriceDecreasing(closingPrices, minDaysDecreasing) && assets.Positions.TryGetValue(symbol, out var position))
+            {
+                if (position.AvailableQuantity > 0)
+                    sellActions.Add(TradingAction.MarketSell(symbol, position.AvailableQuantity, _clock.UtcNow));
+            }
+            else if (IsPriceIncreasing(closingPrices, minDaysIncreasing))
+            {
+                decimal growthRate = CalculateAverageGrowthRate(closingPrices);
+                var lowPrice0 = prediction.Value.Prices.Select(dailyPrice => dailyPrice.LowPrice).ToList()[0];
+                var lowPrice1 = prediction.Value.Prices.Select(dailyPrice => dailyPrice.LowPrice).ToList()[1];
+                var buyLimitPrice = lowPrice0 + (lowPrice1 - lowPrice0) / 2;
+                if(currentPrice != null)
+                {
+                    var tomorrowLowPrice = lowPrice0;
+                    if (tomorrowLowPrice > currentPrice.Low)
+                        buyLimitPrice = currentPrice.Low + (tomorrowLowPrice - currentPrice.Low) / 2;
+                    else
+                        buyLimitPrice = tomorrowLowPrice + (currentPrice.Low - tomorrowLowPrice) / 2;
+                }
+
+                growthRates.Add(new averageGrowthRate(symbol, growthRate, buyLimitPrice));
+            }
+        }
+
+        tradingActions.AddRange(sellActions);
+        tradingActions.AddRange(GetBuyActions(growthRates, cashAvailable, maxBuyCount, topGrowingSymbolsBuyRatio));
+
+        return tradingActions;
+    }
+
+    private async Task<DailyTradingData?> GetCurrentPrice(TradingSymbol symbol, CancellationToken token = default)
+    {
+        var today = DateOnly.FromDateTime(_clock.UtcNow.UtcDateTime);
+        var result = await _marketDataSource.GetDataForSingleSymbolAsync(symbol, today, today, token);
+
+        return (result == null) ? null : result.First();
+    }
+
+    private static bool IsPriceDecreasing(IReadOnlyList<decimal> closingPrices, int maxDaysDecreasing)
+    {
+        var isPriceDecreasing = true;
+        maxDaysDecreasing = maxDaysDecreasing > closingPrices.Count - 1 ? closingPrices.Count - 1 : maxDaysDecreasing;
+
+        for (int i = 1; i <= maxDaysDecreasing; i++)
+        {
+            if (closingPrices[i] > closingPrices[0])
+            {
+                isPriceDecreasing = false;
+                break;
+            }
+        }
+
+        return isPriceDecreasing;
```

Comment:
We could do something like `closingPrices.Take(maxDaysDecreasing).Skip(1).All(price => price <= closingPrices[0])`. But we can leave it like that - it doesn't really change anything.